python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,346 @@
+            Either a list of single band objects or an `OrderedDict` with
+            filter names as keys and single band objects as values, or
+            an instance of an object that inherits from `MultibandBase`.
+        slices: list
+            Slices used to extract a subset of an image
+            (used when `singles` inherits from `MultibandBase` to
+            create a new object with a slice of the original).
+        """
+        err = "Inherited classes from `MultibandBase` require an `__init__` method"
+        raise NotImplementedError(err)
+
+    @property
+    def filters(self):
+        """List of filter names for the single band objects
+        """
+        return self._filters
+
+    @property
+    def singles(self):
+        """List of afw single band objects
+        """
+        return self._singles
+
+    @property
+    def bbox(self):
+        """Bounding box
+        """
+        return self._bbox
+
+    def getBBox(self):
+        """Get the bounding box
+        """
+        return self._bbox
+
+    @property
+    def XY0(self):
+        """Minimum coordinate in the bounding box
+        """
+        return self._xy0
+
+    def setXY0(self, xy0):
+        """Update the XY0 position for each single band object
+        """
+        assert isinstance(xy0, Point2I)
+        self._xy0 = xy0
+        for n in range(len(self.singles)):
+            self.singles[n].setXY0(xy0)
+
+    @property
+    def x0(self):
+        """X0
+
+        X component of XY0 `Point2I.getX()`
+        """
+        return self.XY0.getX()
+
+    @property
+    def y0(self):
+        """Y0
+
+        Y component of XY0 `Point2I.getY()`
+        """
+        return self.XY0.getY()
+
+    @property
+    def yx0(self):
+        """Minimum (y,x) position
+        """
+        return (self.y0, self.x0)
+
+    @property
+    def xy0(self):
+        """Minimum (x,y) position
+        """
+        return (self.x0, self.y0)
+
+    @property
+    def width(self):
+        """Width of the images
+        """
+        return self.XY0.getWidth()
+
+    @property
+    def height(self):
+        """Height of the images
+        """
+        return self.XY0.getHeight()
+
+    def __getitem__(self, args):
+        """Get a slice of the underlying array
+
+        If there is only a single slice, which is a string,
+        then the element of `singles` for the given filter is
+        returned.
+        """
+        if np.issubdtype(type(args), np.integer) or isinstance(args, slice):
+            slices = (args,)
+        else:
+            slices = args
+        if isinstance(slices[0], str):
+            idx = self.filters.index(slices[0])
+            return self.singles[idx]
+        return type(self)(self, slices=slices)
+
+    def __repr__(self):
+        result = "<{0}, filters={1}, yx0={2}, image shape={3}>".format(
+            self.__class__.__name__, self.filters, self.yx0, self.array.shape[-2:])
```

Comment:
Suggest showing the bbox here instead of `yx0` and shape; LSST convention is (x, y) ordering, even if we need to support the transposed version for NumPy interoperability in some places.